app/code/Magento/Wishlist/Block/Customer/Wishlist.php

```diff
@@ -78,14 +83,66 @@ protected function _prepareCollection($collection)
     }
 
     /**
-     * Preparing global layout
+     * Paginate Wishlist Product Items collection
      *
      * @return void
      */
+    public function paginateCollection()
+    {
+        $page = $this->getRequest()->getParam("p", 1);
+        $limit = $this->getRequest()->getParam("limit", 10);
+        $this->_collection
+            ->setPageSize($limit)
+            ->setCurPage($page);
+    }
+
+    /**
+     * Retrieve Wishlist Product Items collection
+     *
+     * @return \Magento\Wishlist\Model\ResourceModel\Item\Collection
+     */
+    public function getWishlistItems()
+    {
+        if ($this->_collection === null) {
+            $this->_collection = $this->_createWishlistItemCollection();
+            $this->_prepareCollection($this->_collection);
+            $this->paginateCollection();
+        }
+
+        return $this->_collection;
+    }
+
+    /**
+     * Preparing global layout
+     *
+     * @return $this
+     */
     protected function _prepareLayout()
     {
+        $this->getWishlistItems();
```

Comment:
Please do not call getters if you do not need the returned result, see my code suggestion below

Reply:
done